### PR TITLE
KIALI-1262 Link to the gateway

### PR DESCRIPTION
**Pull request by aljesusg**
JIRA: [KIALI-1262](https://issues.jboss.org/browse/KIALI-1262)

screenshot from 2018-11-19 13-48-05

### CODE REVIEW

**Comment by aljesusg** [comment]
@hhovsepy do you know the nick of sunil ? He is the asigneed to this task, It's related with QE JIRA [KIALI-1899](https://issues.jboss.org/browse/KIALI-1899). Thanks

**Comment by hhovsepy** [comment]
@skondkar

**Comment by skondkar** [comment]
The Gateway link always points to istio-system namespace and displays UI error for gateways in other namespace.

Example:
There is a gateway 'bookinfo-vs' in bookinfo namespace for which the gateway link dispalys below: 
https://kiali-istio-system.istiooc.jonqe.lab.eng.bos.redhat.com/console/namespaces/istio-system/istio/gateways/bookinfo-gateway

On clicking above link for 'bookinfo-vs', it displays error in UI 
Could not fetch IstioConfig details, Error: [ gateways.networking.istio.io "bookinfo-gateway" not found ]

I installed helloworld service in istio-system namespace and when navigated to the helloworld virtual service->Overview tab, it displayed below link and navigated to the istio config of the helloworld-gateway correctly.
https://kiali-istio-system.istiooc.jonqe.lab.eng.bos.redhat.com/console/namespaces/istio-system/istio/gateways/helloworld-gateway

So the Gateway link works if gateway is in istio-system namespace, however does not work if gateway is in other namespace

**Comment by xeviknal** [comment]
@aljesusg It looks good to me :) One question though, have you tried this DetailObject component in the scenario where there is a recursive call? In that scenario, the link shouldn't be displayed right?

**Comment by aljesusg** [comment]
@skondkar there is not a istio object of type gateway called bookinfo-gateway in that environment in the namspeace istio-system. 

If you check https://kiali-istio-system.istiooc.jonqe.lab.eng.bos.redhat.com/console/istio?namespace=istio-system&istiotype=Gateway there is not a istio object of type gateway called bookinfo-gateway in that environment. the item is in the ns bookinfo https://kiali-istio-system.istiooc.jonqe.lab.eng.bos.redhat.com/console/namespaces/bookinfo/istio/gateways/bookinfo-gateway

If you go for example to the VS https://kiali-istio-system.istiooc.jonqe.lab.eng.bos.redhat.com/console/namespaces/bookinfo/services/productpage?virtualservice=bookinfo-vs&detail=overview there is not link to the gateway, Did you deploy this PR?

File: src/pages/ServiceDetails/ServiceInfo/IstioObjectDetails/VirtualServiceDetail.tsx [hunk 1]
```diff
@@ -87,7 +87,11 @@ class VirtualServiceDetail extends React.Component<VirtualServiceProps> {
           undefined
         )}
         {virtualService.gateways && virtualService.gateways.length > 0 ? (
-          <DetailObject name="Gateways" detail={virtualService.gateways} />
+          <DetailObject
+            name="Gateways"
+            detail={virtualService.gateways}
+            href={`/namespaces/istio-system/istio/gateways`}
```
Comment (xeviknal) [inline src/pages/ServiceDetails/ServiceInfo/IstioObjectDetails/VirtualServiceDetail.tsx]: At the point, the URL has to be built dynamically. Check this [out](https://istio.io/docs/reference/config/istio.networking.v1alpha3/#VirtualService) so see how the gateways can be specified in that field.

**Comment by FilipB** [comment]
kialiqeci

**Comment by rhqci** [comment]
Jenkins CI: kiali-ui-pr-UI-test [#363](https://paste.fedoraproject.org/paste/XhvsLfGwVkHIituXRr4g5w)
  * :red_circle: python-ui-tests-parallel [#[261]](https://paste.fedoraproject.org/paste/40NthZwLdO5mSfr5bqmAvQ)

**Comment by FilipB** [comment]
kialiqeci

**Comment by aljesusg** [comment]
@skondkar @xeviknal It's ready thanks

**Comment by skondkar** [comment]
Verified. The Gateway link works as expected. Verified for bookinfo and istio-system namespaces. The Gateway link navigates correctly to the Istio Config of the Gateway object. Also verified that the breacrumb is correct. Thanks

**Comment by xeviknal** [comment]
@aljesusg there is one more scenario to take care of. In that specific field, istio has a reserved word for referring that this VR has impact to the whole service mesh. The reserved word is `mesh`. 
Therefore, following the current structure we would be linking mesh to an inexistent page: `/gateways/mesh`.
See the reserved word `mesh`definition in the table: https://istio.io/docs/reference/config/istio.networking.v1alpha3/#VirtualService

**Comment by aljesusg** [comment]
Hooray !!! Thanks @xeviknal

**Pull request merged**